src/app/static/src/app/components/downloadResults/UploadModal.vue

```diff
@@ -147,9 +155,28 @@
             setDefaultCheckedItems: function () {
                 this.uploadFilesToAdr = [...outputFileTypes, ...inputFileTypes]
                     .filter(key => this.uploadableFiles.hasOwnProperty(key))
+            },
+            //show and upload only successfully downloaded outputFiles in upload modal
+            getAvailableOutputFiles: function (key) {
+                let availableOutputFiles = {} as UploadFile
+                if (key === "outputZip" && !this.spectrumIsAvailableForUpload) {
+                    availableOutputFiles = this.uploadableFiles[key]
+                } else if (key === "outputSummary" && !this.summaryIsAvailableForUpload) {
+                    availableOutputFiles = this.uploadableFiles[key]
+                }
+
+                return availableOutputFiles
+            },
+            getSectionHeadings: function (sectionIndex) {
+                const hasOutputFile = Object.keys(this.uploadFileSections[0]).length > 0
+                return sectionIndex === 0 ? hasOutputFile ? 'outputFiles' : '' : 'inputFiles';
```

Comment:
Nested ternaries are pretty nasty to read - could add some formatting or parentheses to make it easier?